src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+            string clientId = context.ClientId;
+            string hash = queueMessage.TransactionHash;
+            string cashOperationId = context.CashOperationId;
+
+            await _cashOperationsRepositoryClient.UpdateBlockchainHashAsync(clientId, cashOperationId, hash);
+
+            return true;
+        }
+
+        private async Task<bool> ProcessOutcomeOperation(ProcessCoinEventCommand queueMessage)
+        {
+            var transferTx = await _ethereumTransactionRequestRepository.GetAsync(Guid.Parse(queueMessage.OperationId));
+            CashOutContextData context = await _transactionService.GetTransactionContext<CashOutContextData>(queueMessage.OperationId);
+
+
+            if (transferTx != null)
+            {
+                switch (transferTx.OperationType)
+                {
+                    case OperationType.CashOut:
+                        await SetCashoutHashes(transferTx, queueMessage.TransactionHash);
+                        break;
+                    case OperationType.Trade:
+                        await SetTradeHashes(transferTx, queueMessage.TransactionHash);
+                        break;
+                    case OperationType.TransferToTrusted:
+                    case OperationType.TransferFromTrusted:
+                        await SetTransferHashes(transferTx, queueMessage.TransactionHash);
+                        break;
+                }
+
+                return true;
+            }
+
+            if (context != null)
+            {
+                string clientId = context.ClientId;
+                string hash = queueMessage.TransactionHash;
+                string cashOperationId = context.CashOperationId;
+
+                await _cashOperationsRepositoryClient.UpdateBlockchainHashAsync(clientId, cashOperationId, hash);
+            }
+
+
+            return true;
+        }
+
+        //TODO: Split wth the help of the process management
+        private async Task<bool> ProcessFailedCashout(ProcessCoinEventCommand queueMessage, IEventPublisher eventPublisher)
+        {
+            CashOutContextData context = await _transactionService.GetTransactionContext<CashOutContextData>(queueMessage.OperationId);
+
+            if (context != null)
+            {
+                string transactionHandlerUserId = "auto redeem";
+                string clientId = context.ClientId;
+                string hash = queueMessage.TransactionHash;
+                string cashOperationId = context.CashOperationId;
+                string assetId = context.AssetId;
+                var amount = context.Amount;
+
+                try
+                {
+                    var asset = await _assetsService.AssetGetAsync(assetId);
+                    await _cashOperationsRepositoryClient.UpdateBlockchainHashAsync(clientId, cashOperationId, hash);
+                    var pt = await _paymentTransactionsRepository.TryCreateAsync(PaymentTransaction.Create(hash,
```

Comment:
Add ChaosKitty before and after this call

Reply:
fixed